src/simplify_reshapes.cpp

```diff
@@ -949,6 +949,43 @@ struct find_reshape_reshape_dot
     }
 };
 
+// Move shape transformation before pointwise ops when doing pointwise op with constant
+// eg. mul -> add -> transpose -> dot becomes transpose -> mul -> add -> dot
+struct find_pointwise_transpose_dot
+{
+    auto matcher() const
+    {
+        auto reshapes = reshaper_names();
+        reshapes.insert("transpose");
+        auto const_pw = match::pointwise()(
+            match::either_arg(0, 1)(match::is_constant().bind("const"), match::any().bind("inp")));
+        auto match_shape_op = match::name("transpose")(match::args(const_pw.bind("pw")));
+        auto skip_pointwise_reshape_outputs =
+            match::skip_output(match::any_of(match::name(reshapes), match::pointwise()));
```

Comment:
Logical question about `skip_output` does it only skip one time or one op or can it skip multiple ops ?

e.g. 
if 
let's say graph is `mul --> add -- sub --> transpose --> dot` then after two applications of this matcher it would become 

`mul --> transpose --> add --> sub --> dot` 

can `skip_output` skip two pointwises and match with `dot` in this case ?

Reply:
Thats a good point actually, my code is assuming that it can skip multiple times. @pfultz2 is this true?

Reply:
actually it has to be working that way, otherwise the test case wouldn't pass. Test case only runs the pass once and the transpose moves before both mul and add

Reply:
>  Test case only runs the pass once and the transpose moves before both mul and add

The pass itself is ran 4 times.

Reply:
Oh right thats true. But in any case what I meant is that it wouldn't match in the second iteration if `skip_ouput` didnt skip multiple instructions. Just want to make sure Im understanding that matcher correctly

Reply:
> actually it has to be working that way, otherwise the test case wouldn't pass. Test case only runs the pass once and the transpose moves before both mul and add

which test case are you referring to ? test case i see only has two pointwise e.g mul and add

Reply:
> Thats a good point actually, my code is assuming that it can skip multiple times. @pfultz2 is this true?

Yes this is true, its called recursively in the matcher:

https://github.com/ROCm/AMDMIGraphX/blob/f522bb9d41a3ee11124c02babbb510b6274ea41e/src/include/migraphx/matcher.hpp#L671